### PR TITLE
[fix][metrics] fixed a partitioned publisher topic stat aggregation bug

### PR DESCRIPTION
**Pull request by heesung-sohn**
Fixes #xyz

Master Issue: #xyz

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [x] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java [hunk 1]
```diff
@@ -154,8 +155,8 @@ public NonPersistentTopicStatsImpl add(NonPersistentTopicStats ts) {
         Objects.requireNonNull(stats);
         super.add(stats);
         this.msgDropRate += stats.msgDropRate;
-
-        stats.getNonPersistentPublishers().forEach(s -> {
+        IntStream.range(0, stats.getNonPersistentPublishers().size()).forEach(index -> {
+            NonPersistentPublisherStats s = stats.getPublishers().get(index);
```
Comment (eolivelli) [inline pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java]: should this be "getNonPersistentPublishers" ?

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java [hunk 2]
```diff
@@ -154,8 +155,8 @@ public NonPersistentTopicStatsImpl add(NonPersistentTopicStats ts) {
         Objects.requireNonNull(stats);
         super.add(stats);
         this.msgDropRate += stats.msgDropRate;
-
-        stats.getNonPersistentPublishers().forEach(s -> {
+        IntStream.range(0, stats.getNonPersistentPublishers().size()).forEach(index -> {
```
Comment (eolivelli) [inline pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java]: if we care about memory allocations and CPU cycles, I suggest to use a simple for `(int index=0; index < ...)` loop.

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java (hunk 1 above)
Comment (heesung-sohn) [reply pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java]: Updated to be consistent, though they are identical.

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java (hunk 2 above)
Comment (heesung-sohn) [reply pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java]: updated.

**Pull request closed without merging**